src/nevermined/api/AssetsApi.ts

```diff
@@ -208,18 +212,30 @@ export class AssetsApi extends RegistryBaseApi {
   ): Promise<string | true> {
     const ddo = await this.resolve(did)
     const { attributes } = ddo.findServiceByType('metadata')
-    const { serviceEndpoint, index } = ddo.findServiceByType('access')
+    let service
+    if (typeof serviceReference === 'number') {
+      service = ddo.findServiceById(serviceReference)
+    } else {
+      service = ddo.findServiceByType(serviceReference)
```

Comment:
not sure about this one. here we return the first service which may not be the intended service. I think we should stick to `findServiceById`

Reply:
This is for keeping backward compatibility with the existing API. If we remove the `findServiceByType` we must enforce using the serviceIndex everywhere so that parameter is not optional anymore and has to be given in all the API calls. 
IMO taking into account we come from a situation where only one service of the same type is supported, a simpler migration would be to keep the API backwards compatible.